app/views/portal/show.rb

```diff
@@ -783,154 +753,173 @@ def iiif_manifesto(identifier, collection)
+    end
 
-        # disable play / download
+    def media_items
+      items = salient_web_resources.map do |web_resource|
+        web_resource_media_item(web_resource)
+      end
 
-        if mime_type == 'video/mpeg'
-          item[:playable] = false
-        end
+      {
+        required_players: item_players(items),
+        single_item: items.size == 1,
+        empty_item: items.size == 0,
+        items: items,
+        more_thumbs_url: request.original_url.split('.html')[0] + '/thumbnails.json'
+      }
+    end
 
-        if mime_type == 'audio/flac'
-          item[:playable] = true
-        end
+    def salient_web_resources
+      aggregation = document.aggregations.first
+      return [] unless aggregation.respond_to?(:webResources)
 
-        if media_type == 'text' && (mime_type == 'text/plain; charset=utf-8' || !mime_type)
-          item[:playable] = false
-          item[:downloadable] = false
-        end
+      view_urls = aggregation.fetch('hasView', []) + [aggregation.fetch('edmIsShownBy', nil)]
+      web_resources = aggregation.webResources.dup
+      edm_web_resource = web_resources.find { |web_resource| render_document_show_field_value(web_resource, 'about') == edm_resource_url }
+      # make sure the edm_is_shown_by is the first item
+      web_resources.unshift(web_resources.delete(edm_web_resource)) unless edm_web_resource.nil?
+      web_resources.select! { |wr| view_urls.compact.include?(wr.fetch('about', nil)) }
+      web_resources.uniq { |wr| wr.fetch('about', nil) }
+    end
 
-        if media_type == 'text' && mime_type == 'text/plain; charset=utf-8'
-          item[:playable] = false
-          item[:downloadable] = false
-        end
+    def item_technical_metadata(web_resource, mime_type)
+      file_size = number_to_human_size(render_document_show_field_value(web_resource, 'ebucoreFileByteSize')) || ''
+      {
+        mime_type: mime_type,
+        file_size: file_size.split(' ').first,
+        file_unit: file_size.split(' ').last,
+        codec: render_document_show_field_value(web_resource, 'edmCodecName'),
+        width: render_document_show_field_value(web_resource, 'ebucoreWidth'),
+        height: render_document_show_field_value(web_resource, 'ebucoreHeight'),
+        size_unit: 'pixels',
+        runtime: render_document_show_field_value(web_resource, 'ebucoreDuration'),
+        runtime_unit: 'seconds'
+      }
+    end
 
-        if media_type == 'video' && mime_type == 'text/plain; charset=utf-8'
-          item[:playable] = false
-          item[:downloadable] = false
-        end
+    def item_playable?(web_resource_url, mime_type, media_type)
+      if web_resource_url.blank? ||
+        mime_type.blank? ||
+        (mime_type == 'video/mpeg') ||
+        (media_type == 'text' && mime_type == 'text/plain; charset=utf-8') ||
+        (media_type == 'video' && mime_type == 'text/plain; charset=utf-8')
```

Comment:
Use 4 (not 2) spaces for indenting a condition in an `if` statement spanning multiple lines.